src/neo/SmartContract/Native/NeoToken.cs

```diff
@@ -56,6 +57,73 @@ public sealed class NeoToken : FungibleToken<NeoToken.NeoAccountState>
         internal NeoToken()
         {
             this.TotalAmount = 100000000 * Factor;
+
+            var events = new List<ContractEventDescriptor>(Manifest.Abi.Events)
+            {
+                new ContractEventDescriptor
+                {
+                    Name = "CandidateRegistered",
+                    Parameters = new ContractParameterDefinition[]
+                    {
+                        new ContractParameterDefinition()
+                        {
+                            Name = "pubkey",
+                            Type = ContractParameterType.PublicKey
+                        },
+                        new ContractParameterDefinition()
+                        {
+                            Name = "votes",
+                            Type = ContractParameterType.Integer
+                        }
+                    }
+                },
+                new ContractEventDescriptor
+                {
+                    Name = "CandidateUnregistered",
```

Comment:
Merge it with `CandidateRegistered`? Maybe `CandidateStateChanged` or something.

Reply:
If we have `Register` and `Unregister`, then we need `Vote` and `Unvote`.

Reply:
I'd be totally fine with that too. However, we do not have `Burn` or `Mint` either. We just have `Transfer` with a `Null` `to` or `from` respectively. It's not like everything needs to be (or even is) symmetrical in the system.

Reply:
Then I prefer to merge Register and Unregister.🤭

Reply:
Unify is always good

Reply:
Unifying just makes post processing "harder" where explicit you can `switch` on the event name. 
 
> Then I prefer to merge Register and Unregister.🤭

I was trying to point out that `vote` could work similar to the `transfer` event.  How about we vote for it 

🎉  for all explicit (CandidateRegistered, CandidateUnregistered, Vote, Unvote)
❤️  for implicit (`CandidateStateChanged` with extra param for `Registered`/`Unregistered` state, `vote` with  no extra parameter and the consumer parses the `to` field to determine if it is a `vote` or `unvote`)

Reply:
My first choice would have been `registerCandidate`/`unregisterCandidate` and  a single `vote`, pretty much how it was PR'ed. But Erik seems to want all explicit or all implicit hence the 2 choices above 🤷‍♂️